ansible/roles/osdsdock/scenarios/ceph.yml

```diff
@@ -13,79 +13,161 @@
 # limitations under the License.
 
 ---
+- name: ceph pre-flight check
+  include_tasks: roles/osdsdock/scenarios/ceph_preflight_check.yml
+
 - name: check ceph exists
-  shell: "{{ item }}"
-  with_items: 
-    - bash ./script/check_ceph_exist.sh
+  shell: which ceph
+  args:
+    executable: /bin/bash
   ignore_errors: true
-  register: cephres
-
-- name: check if ceph.conf folder exist when ceph backend enable
-  shell: ls /etc/ceph
+  register: ceph_existence
+#
+# check installed ceph major version and set it to 'installed_ceph_version'.
+#
+- name: check the installed ceph version
+  shell: ceph --version  | egrep ^ceph | awk '{print $3;}' | cut -d "." -f 1,1
+  args:
+    executable: /bin/bash
   ignore_errors: true
   register: result
+  when: ceph_existence["rc"] == 0
 
-- name: create ceph.conf folder if not exist when ceph backend enable
-  file:
-    path: "{{ item }}"
-    state: directory
-    mode: 0755
-  with_items:
-    - "{{ ceph_backend_conf }}"
-  when: cephres["results"][0]["stdout"] == "HEALTH_OK" and result["rc"] != 0
+- name: set_fact installed_ceph_version
+  set_fact:
+    installed_ceph_version: "{{ result['stdout']  }}"
+  when:
+    - ceph_existence["rc"] == 0
+#
+# check 'ceph health' and set the result into 'ceph_health'.
+#
+- name: check if the installed ceph is healthy or not
+  shell: |
+    ceph health
+  args:
+    executable: /bin/bash
+  register: result
+  ignore_errors: true
+  when: ceph_existence["rc"] == 0
+
+- name: set_fact ceph_health
+  set_fact:
+    # workaround
+    ceph_health: "{{ result['stdout'] }}{{ result['stderr'] }}"
+    when: ceph_existence["rc"] == 0
```

Comment:
Please remove whitespace

Reply:
Following our discussion at slack, I deleted 2 white spaces in front of  'when:' at line 57.

Reply:
Thank you